app/src/main/java/ca/dal/cs/athletemonitor/athletemonitor/ExerciseActivity.java

```diff
@@ -0,0 +1,95 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_exercise);
+
+        // Click listener for create new exercise button
+        findViewById(R.id.createExerciseButton).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent createExerciseActivityIntent = new Intent(ExerciseActivity.this, CreateExerciseActivity.class);
+                createExerciseActivityIntent.putExtras(getIntent().getExtras());
+                startActivity(createExerciseActivityIntent);
+            }
+        });
+
+        AccountManager.getUser(getIntent().getExtras().getString("username"), new AccountManager.UserObjectListener() {
+            @Override
+            public void onUserPopulated(User user) {
+                // Get the layout to add exercises to
+                LinearLayout layout = findViewById(R.id.exerciseLinearLayout);
+
+                boolean alternateColor = false;
+
+                // Get the user's list of exercises
+                List<Exercise> exercises = user.getUserExercises();
+                // Iterate and add exercises to screen
+                for (Exercise exercise : exercises) {
+                    // Build a new TextView for this exercise
+                    TextView exerciseText = new TextView(ExerciseActivity.this);
+                    exerciseText.setText(exercise.getName());
+                    exerciseText.setTextSize(28);
+                    exerciseText.setPadding(0, 30, 0, 30);
+                    if (alternateColor) exerciseText.setBackgroundColor(Color.LTGRAY);
+                    exerciseText.setTextAlignment(View.TEXT_ALIGNMENT_CENTER);
+                    LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
+                            LinearLayout.LayoutParams.MATCH_PARENT,
+                            LinearLayout.LayoutParams.WRAP_CONTENT);
+                    params.setMargins(0, 4, 0, 0);
+                    exerciseText.setLayoutParams(params);
+
+                    // Add a click listener to show more information
+                    exerciseText.setOnClickListener(new DialogOnClickListener(exercise));
+                    // Add the text view to the screen
+                    layout.addView(exerciseText);
+
+                    alternateColor = !alternateColor;
+                }
+            }
+        });
+    }
+
+    // Custom click listener implementation, so that we can access the exercise data.
+    class DialogOnClickListener implements View.OnClickListener{
+        Exercise exercise;
+        public DialogOnClickListener(Exercise exercise){
+            this.exercise = exercise;
+        }
+        @Override
+        public void onClick(View v){
+            AlertDialog.Builder builder = new AlertDialog.Builder(ExerciseActivity.this);
+            LayoutInflater inflater = ExerciseActivity.this.getLayoutInflater();
```

Comment:
`ExerciseActivity.this.getLayoutInflater()` Is this notation required because you are in callback?

Reply:
Yes, inside of the callback (and outside of an Activity class in general) you refer to the activity instance as Activity.this. 

(I actually just removed this line because it was left over from somewhere else)